frontend/packages/console-shared/src/hooks/useCanClusterUpgrade.ts

```diff
@@ -0,0 +1,21 @@
+import { useAccessReviewAllowed } from '@console/dynamic-plugin-sdk';
+import { ClusterVersionModel } from '@console/internal/models';
+import { ClusterVersionKind, hasAvailableUpdates } from '@console/internal/module/k8s';
+
+export const isClusterExternallyManaged = () => {
+  return window.SERVER_FLAGS.controlPlaneTopology === 'External';
+};
+
+export const useCanClusterUpgrade = (clusterVersion: ClusterVersionKind): boolean => {
+  const hasPermissionsToUpdate = useAccessReviewAllowed({
+    group: ClusterVersionModel.apiGroup,
+    resource: ClusterVersionModel.plural,
+    verb: 'patch',
+    name: 'version',
+  });
+  const notExternallyManaged = !isClusterExternallyManaged();
+  const bandingNotDedicated = window.SERVER_FLAGS.branding !== 'dedicated';
```

Comment:
OSD should not be upgradable as well?

Reply:
@jhadvig I didn't come up with the logic, just moved it. It was already here: https://github.com/openshift/console/pull/11339/files#diff-0fd9d8f08b4c2310a37d6a88fab30d2e4fc9eea2854e85f77992a6740f61124fL43-L49